apps/prairielearn/src/models/variant.ts

```diff
@@ -239,7 +239,7 @@ export async function selectAndAuthzVariant(options: {
       !is_administrator &&
       (!authnHasCourseInstancePermissionView || !hasCourseInstancePermissionView)
     ) {
-      denyAccess();
+      denyAccess('Access denied (must have student data viewer permissions)');
```

Comment:
Drive-by improvement: we can share a slightly better error message in this case.